docs/concurrency-adoption-guidelines.md

```diff
@@ -0,0 +1,179 @@
+# Swift Concurrency adoption guidelines for Swift Server Libraries
```

Comment:
may be nice to add an "index" at the end before we merge so people can see the high level outline/jump to particular sections as this is getting rather lengthy

Reply:
Good point, will do